### PR TITLE
Fix #2782

### LINKED ISSUE #2782: Cannot emit sRGB-chunk
### What did you do?
I tried to add an sRGB-chunk via PngInfo to the PNG image. (Script at the bottom).

### What did you expect to happen?
```
File: /mnt/c/Users/stuxc/Desktop/Download_srgb3.png (1045 bytes)
  chunk IHDR at offset 0x0000c, length 13
    640 x 480 image, 24-bit RGB, non-interlaced
  chunk sRGB at offset 0x00025, length 1
    rendering intent = absolute colorimetric
  chunk gAMA at offset 0x00032, length 4: 0.45455
  chunk cHRM at offset 0x00042, length 32
    White x = 0.3127 y = 0.329,  Red x = 0.64 y = 0.33
    Green x = 0.3 y = 0.6,  Blue x = 0.15 y = 0.06
  chunk IDAT at offset 0x0006e, length 915
    zlib: deflated, 32K window, default compression
  chunk IEND at offset 0x0040d, length 0
No errors detected in /mnt/c/Users/stuxc/Desktop/Download_srgb3.png (6 chunks, 99.9% compression).
```
Note the sRGB chunk at offset 0x00025

### What actually happened?
```
File: /mnt/c/Users/stuxc/Desktop/Download_srgb5.png (1032 bytes)
  chunk IHDR at offset 0x0000c, length 13
    640 x 480 image, 24-bit RGB, non-interlaced
  chunk gAMA at offset 0x00025, length 4: 0.45455
  chunk cHRM at offset 0x00035, length 32
    White x = 0.3127 y = 0.329,  Red x = 0.64 y = 0.33
    Green x = 0.3 y = 0.6,  Blue x = 0.15 y = 0.06
  chunk IDAT at offset 0x00061, length 915
    zlib: deflated, 32K window, default compression
  chunk IEND at offset 0x00400, length 0
No errors detected in /mnt/c/Users/stuxc/Desktop/Download_srgb5.png (5 chunks, 99.9% compression).
```

Both outputs were created by `pngcheck -vt <Filename>`

### What versions of Pillow and Python are you using?
`Pillow==4.3.0` affected code is also in the current master branch.
Python 3.6.2 using Windows Subsystem for Linux (Windows 10 Build 15063)

I think the error is located at:
https://github.com/python-pillow/Pillow/blob/master/PIL/PngImagePlugin.py#L732
This removes the sRGB-chunk from the list of valid chunks if (and only if) there is no iCCP-chunk given to the save function.

### Script that replicates the bug
```python
import struct
from PIL import Image
from PIL.PngImagePlugin import PngInfo

GAMA_SRGB = struct.pack('!I', 45455)
CHRM_SRGB = struct.pack("!8I", 31270, 32900, 64000, 33000, 30000, 60000, 15000, 6000)

SRGB_PNGINFO = PngInfo()
SRGB_PNGINFO.add(b"sRGB", b'\3')
SRGB_PNGINFO.add(b"gAMA", GAMA_SRGB)
SRGB_PNGINFO.add(b"cHRM", CHRM_SRGB)

image = Image.new('RGB', (640, 480))
with open("Download_srgb5.png", "wb") as f:
    image.save(f, format="png", compress_level=6, pnginfo=SRGB_PNGINFO)
```

### Significance
Chrome 61 expects images to have a color profile embedded in them. In my case, it is of vital importance that the generated images are being shown using the sRGB color profile.

### PR DESCRIPTION
**Pull request by stuxcrystal**
Fixes #2782

### CODE REVIEW

File: PIL/PngImagePlugin.py [hunk 1]
```diff
@@ -728,7 +728,9 @@ def _save(im, fp, filename, chunk=putchunk, check=0):
         name = b"ICC Profile"
         data = name + b"\0\0" + zlib.compress(icc)
         chunk(fp, b"iCCP", data)
-    else:
+        
+        # You must either have sRGB or iCCP.
+        # Disallow sRGB chunks when a iCCP-chunk has been emitted.
```
Comment (radarhere) [inline PIL/PngImagePlugin.py]: an?

**Comment by homm** [comment]
Tests?

**Comment by hugovk** [comment]
A couple of general points: please use more descriptive commit messages (and PR titles), and include unit tests with the fix. Thank you!

**Comment by wiredfool** [comment]
Superseded by #2898

**Pull request closed without merging**